pkg/mssqlcontainer/ingest/mechanism/attach.go

```diff
@@ -0,0 +1,65 @@
+package mechanism
+
+import (
+	"fmt"
+	"github.com/microsoft/go-sqlcmd/internal/container"
+)
+
+type attach struct {
+	controller  *container.Controller
+	containerId string
+}
+
+func (m *attach) Initialize(controller *container.Controller) {
+	m.controller = controller
+}
+
+func (m *attach) CopyToLocation() string {
+	return "/var/opt/mssql/data"
+}
+
+func (m *attach) Name() string {
+	return "attach"
+}
+
+func (m *attach) FileTypes() []string {
+	return []string{"mdf"}
+}
+
+func (m *attach) BringOnline(databaseName string, containerId string, query func(string), options BringOnlineOptions) {
+	text := `SET NOCOUNT ON; `
+
+	m.containerId = containerId
+	m.setFilePermissions(m.CopyToLocation() + "/" + options.Filename)
+	if options.LdfFilename == "" {
+		text += `CREATE DATABASE [%s] ON (FILENAME = '%s/%s') FOR ATTACH;`
+		query(fmt.Sprintf(
```

Comment:
>query(fmt.Sprintf [](http://example.com/codeflow?start=2&length=17)

you need to escape the parameters properly, and use nvarchar for the file name

eg from SMO
```C#
ddl.AppendFormat(SmoApplication.DefaultCulture, ", FILENAME = N'{0}' ", SqlString(fileName));

        public static String SqlString(String s)
        {
            return EscapeString(s, '\'');
        }
```

Reply:
it'd be a good test case to have some file names that use extended characters. Go stores strings as UTF8 but nvarchar is UTF16, while varchar is SQL-collation-dependent. 
How do we guarantee the Go strings get encoded properly in the query?

Reply:
maybe go-mssqldb does the right thing already for strings and we should use a parameterized query to let the driver do the conversion instead of manually inserting tokens in the query.